app/models/activity.rb

```diff
@@ -26,6 +26,11 @@ def project=(new_project); end
   scope :latest, -> do
     includes(:trackable).order('activities.created_at DESC').limit(20)
   end
+  scope :filter_by_user_id, -> (id) { where user_id: id }
+  scope :filter_by_type, -> (type) { where trackable_type: type }
+  scope :filter_by_date, -> (period_start, period_end) {
+    where('created_at >= ? AND created_at <= ?', period_start.beginning_of_day, period_end.end_of_day)
```

Comment:
What about specifying `beginning_of_day` and `end_of_day` in the controller instead?